sci-libs/vtk/vtk-9.0.3-r4.ebuild

```diff
@@ -79,7 +79,7 @@ RDEPEND="
 	json? ( dev-libs/jsoncpp:= )
 	mpi? (
 		sci-libs/h5part
-		sys-cluster/openmpi[cxx,romio]
+		virtual/mpi[cxx,romio]
```

Comment:
Could you please try, if using `virtual/mpi[cxx(-),romio(-)]` here solves the issue with `sys-cluster/nullmpi`? The `virtual/mpi` package uses 4-way USEDEP strings, and nullmpi doesn't have the `cxx` and `romio` USE flags. So if we explicitly request those, this could be the source of this issue.

Reply:
I made the change, but I get the same problem.

Reply:
I tried installing nullmpi and virtual/mpi (all USE flags off) manually, but `sci-libs/hdf5` pulls in `virtual/mpi[romio]` which then will uninstall nullmpi and pull in mpich instead. Why it's selecting mpich, althoug openmpi is the first entry in the any-of DEPSTRING in virtual/mpi I do not know.

Do you have `fortran` USE flag enabled on virtual/mpi? If so, this is likely the cause for it. If you switch it off, I found the dependency tree could successfully be resolved, with the restriction, I told in the previous paragraph.

So, because of the mpi dependency of hdf5, it's not possible to use nullmpi. I'm only wondering, whether we should just ignore it or explicitly don't depend on it.

Reply:
Strange. I have `>=virtual/mpi-2.0-r6 -romio -cxx -threads nullmpi -fortran` and still get the same problem.

Reply:
> Sorry, I don't understand. We are trying to test `nullmpi` specifically and replacement of `nullmpi` with `mpich` is not what we want, right? I think all implementations except `nullmpi` work well with or without `fortran` use flag.

No, we don't want this, but it's unavoidable. HDF5, a hard dependency of vtk (if we don't want to use the bundled version that is) depends on `virtual/mpi[romio]`. Because nullmpi doesn't have this use flag, one of either mpich or openmpi will be pulled in. So, from my understanding, there's no way to build vtk against nullmpi.

Reply:
> I have `>=virtual/mpi-2.0-r6 -romio -cxx -threads nullmpi -fortran` and still get the same problem for `nullmpi`.

The nullmpi USE flag is only used for prefix installations, so it's not needed.

Reply:
Maybe some from @gentoo/sci can bring some light into this?

Reply:
I think nullmpi might be usable on other platforms but it's useless anyway, I think. It's not a "real" MPI implementation and is incomplete. There is no need to worry about building VTK with it.

If it specifically failed to build against it, you can depend on mpi[-nullmpi(-)]. But if you're just wondering if you should bother testing: no, do not test.

Reply:
I haven't tried to build against nullmpi using `ebuild`. With `emerge` it's not possible to test, it will always pull in one of the mpich or openmpi MPI implementations.

Reply:
Ah. Then let's leave it as-is.

Reply:
> If it specifically failed to build against it, you can depend on mpi[-nullmpi(+)].

AFAICS the nullmpi USE flag has nothing to do with the nullmpi package, but instead handles the dependency for native-mpi on prefix installations.